src/Data/TypeRepMap/Internal.hs

```diff
@@ -464,11 +488,29 @@ fromSortedList l = runST $ do
                 writeArray result i (indexArray origin newFirst)
                 loop (2 * i + 2) (newFirst + 1)
 
+-- Returns a list of indexes which represents the "sorted" order of an array generated by
+-- fromSortedList of the provided length.
+-- I.e. fmap (fromSortedList [1, 2, 3, 4, 5] !!) (generateOrderMapping 5) == [1, 2, 3, 4, 5]
```

Comment:
I think, it also will be useful to give a few documentation examples of the `generateOrderMapping` function: what it produces for different outputs.

Also, it looks like you can implement a property-based test for this function using `hedgehog`. Something like:

```haskell
n <- forAll genInt
fmap (fromSortedList [1 .. n] !!) (generateOrderMapping n) === [1 .. n]
```

Never hurts to have more tests about invariants :slightly_smiling_face: